src/signatureUtils.js

```diff
@@ -0,0 +1,227 @@
+            pubKeyUrl1, pubKeyUrl2, recipientClientId, error.message)
+        return genErrorResponse(500, 'Error occurred while fetching Public Key')
+    }
+    var signatures = [digiSignature1, digiSignature2] 
+    var keys = [pubKey1Pem, pubKey2Pem] 
+    return await verifySignatureHelper(signatures, signedPayload, keys, recipientClientId)
+}
+
+/**
+ * Wrapper to fetch the key from aio-lib-state, if not present fetch using
+ * the cloud front url and set in aio-lib-state 
+ * 
+ * @param {*} pubKeyUrl cloud front url of format https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+ * @param {*} state aio-lib-state client
+ * @returns public key
+ */
+async function fetchFromCacheOrApi(pubKeyUrl, state) {
+    var publicKey
+    let publicKeyFileName
+    try {
+        // public key url is the cloud front url in this format https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+        publicKeyFileName = pubKeyUrl.substring(pubKeyUrl.lastIndexOf('/') + 1)
+        publicKey = await state.get(publicKeyFileName)
+    } catch (error) {
+        logger.error('error fetching from aio lib state due to => %s', error.message)
+        throw error
+    }
+
+    if (publicKey) {
+        return publicKey
+    }
+
+    logger.info('public key for url %s not present in aio state lib cache, fetching from api..', pubKeyUrl)
+    // fetch from api and set in cache default expiry 24h
+    publicKey = await fetchPublicKeyFromCloudFront(pubKeyUrl)
+    await state.put(publicKeyFileName, publicKey)
+    return publicKey
+}
+
+/**
+ * Fetches public key using the cloud front public key url
+ * 
+ * @param {*} publicKeyUrl - cloud front public key url of format 
+ * https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+ * @returns public key
+ */
+async function fetchPublicKeyFromCloudFront(publicKeyUrl) {
+    var pubKey
+    await fetch(publicKeyUrl)
+        .then(response => response.text())
+        .then(text => {
+            logger.info('successfully fetched the public key %s from cloud front url %s', text, publicKeyUrl)
+            pubKey = text
+        })
+        .catch(err => {
+            logger.error('error fetching the public key from cloud front url %s due to => %s', publicKeyUrl, err.message)
+            throw err
+        })
+    return pubKey
+}
+
+/**
+ * Digital Signature Verification Helper
+ * 
+ * @param {*} digitalSignature1 - I/O Events generated digital signature1
+ * @param {*} digitalSignature2 - I/O Events generated digital signature2
+ * @param {*} signedPayload - I/O Events proper signed payload
+ * @param {*} pubKey1Pem - I/O Events generated PEM encoded public key1
+ * @param {*} pubKey2Pem - I/O Events generated PEM encoded public key1
+ * @param {*} recipientClientId - target recipient client id
+ * @returns {boolean} true if either signatures are valid or false
+ */
+async function verifySignatureHelper(digitalSignatures, signedPayload, publicKeys, recipientClientId) {
```

Comment:
Why don't we just call this `verifySignature()`?

Reply:
done